### PR TITLE
Implement cupy.show_config and cupyx.get_runtime_info

### PR DESCRIPTION
**Pull request by kmaehashi**
* `cupy.show_config` is a counterpart for `numpy.show_config` that prints configuration information to stdout.
  Note that `cupy.show_config` prints both compile-time and runtime info, whereas `numpy.show_config` only prints compile-time info.
* `cupyx.get_runtime_info` is a hidden API to be used in Chainer (see PR https://github.com/chainer/chainer/pull/4298)

### CODE REVIEW

File: cupy/__init__.py [hunk 1]
```diff
@@ -614,3 +616,8 @@ def get_default_pinned_memory_pool():
 
     """
     return _default_pinned_memory_pool
+
+
+def show_config():
+    """Prints the current runtime configuration to standard output."""
+    sys.stdout.write(str(cupyx.get_runtime_info()))
```
Comment (niboshi) [inline cupy/__init__.py]: How about flushing?

File: cupyx/runtime.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            ('CUDA Driver Version', self.cuda_driver_version),
+            ('CUDA Runtime Version', self.cuda_runtime_version),
+            ('cuDNN Build Version', self.cudnn_build_version),
+            ('cuDNN Version', self.cudnn_version),
+            ('NCCL Build Version', self.nccl_build_version),
+        ]
+        width = max([len(r[0]) for r in records]) + 2
+        fmt = '{:' + str(width) + '}: {}\n'
+        s = six.StringIO()
+        for r in records:
```
Comment (niboshi) [inline cupyx/runtime.py]: It's safer to expand tuples like `for k, v in records:`.

File: tests/cupy_tests/test_init.py [hunk 3]
```diff
@@ -92,6 +92,11 @@ def test_get_default_pinned_memory_pool(self):
         self.assertIsInstance(p, cupy.cuda.pinned_memory.PinnedMemoryPool)
 
 
+class TestShowConfig(unittest.TestCase):
+
+    def test_show_config(self):
+        cupy.show_config()
```
Comment (niboshi) [inline tests/cupy_tests/test_init.py]: What is this test for?

File: cupy/__init__.py (hunk 1 above)
Comment (kmaehashi) [reply cupy/__init__.py]: Fixed.

File: cupyx/runtime.py (hunk 2 above)
Comment (kmaehashi) [reply cupyx/runtime.py]: Fixed.

File: tests/cupy_tests/test_init.py (hunk 3 above)
Comment (kmaehashi) [reply tests/cupy_tests/test_init.py]: Sorry the test was incomplete. Fixed.

**Comment by niboshi** [comment]
LGTM. Please fix style error.

**Comment by kmaehashi** [comment]
Thanks, fixed.

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**